### PR TITLE
Fix nullable motion events

### PR DESCRIPTION
**Pull request by yunikkk**
Also bumped dependencies and used Maps v10 in the test app

Original ticket - https://github.com/mapbox/mapbox-gestures-android/pull/101

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # Changelog for the Mapbox Gestures for Android
 
+## 0.8.0 - 
+#### Breaking changes
```
Comment (kiryldz) [inline CHANGELOG.md:4]: also update readme for the repo with instructions how to setup Maps SDK v10?

File: gradle/dependencies.gradle [hunk 2]
```diff
@@ -1,36 +1,36 @@
 ext {
     androidVersions = [
-            minSdkVersion    : 14,
-            targetSdkVersion : 28,
-            compileSdkVersion: 28,
+            minSdkVersion    : 21,
```
Comment (kiryldz) [inline gradle/dependencies.gradle:3]: deserves changelog entry, doesn't it?

**Comment by pengdev** [comment]
Shall we validate the NonNull annotations for all the changes introduced in https://github.com/mapbox/mapbox-gestures-android/pull/95 ? it might apply to other gesture detector overwrites as well.

File: gradle/wrapper/gradle-wrapper.properties [hunk 3]
```diff
@@ -1,6 +1,6 @@
-#Fri Jan 18 15:22:24 CET 2019
+#Thu Aug 18 15:53:42 EEST 2022
 distributionBase=GRADLE_USER_HOME
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.0.2-bin.zip
```
Comment (kiryldz) [inline gradle/wrapper/gradle-wrapper.properties:3]: deserves changelog entry, doesn't it?

File: gradle/dependencies.gradle (hunk 2 above)
Comment (pengdev) [reply gradle/dependencies.gradle:3]: as a stand-alone library, this minSdkVersion bump seems to be a major breaking change. I'd suggest to keep the same 14 as before.

File: build.gradle [hunk 4]
```diff
@@ -1,26 +1,43 @@
...
         jcenter()
+        mavenCentral()
+        maven {
+            url 'https://api.mapbox.com/downloads/v2/releases/maven'
+            authentication {
+                basic(BasicAuthentication)
+            }
+            credentials {
+                username = "mapbox"
+                password = System.getenv("SDK_REGISTRY_TOKEN") ?: project.property("SDK_REGISTRY_TOKEN") as String
```
Comment (LukasPaczos) [inline build.gradle:14]: Could you add installation instructions to readme? There's already a small section about the token that will have to get refactored now.

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,9 @@
 # Changelog for the Mapbox Gestures for Android
 
+## 0.8.0 -
```
Comment (LukasPaczos) [inline CHANGELOG.md:3]: Missing a date.

**Comment by yunikkk** [comment]
> Shall we validate the NonNull annotations for all the changes introduced in #95 ? it might apply to other gesture detector overwrites as well.

The only detector that passes nullable MotionEvents that were marked NonNull was the SimpleStandardOnGestureListener.
What kind of validation do you have in mind?

File: CHANGELOG.md (hunk 5 above)
Comment (yunikkk) [reply CHANGELOG.md:3]: Yeah, haven't added it since wasn't sure I'll cut a release today ;)

File: build.gradle (hunk 4 above)
Comment (yunikkk) [reply build.gradle:14]: Will do

File: gradle/wrapper/gradle-wrapper.properties (hunk 3 above)
Comment (yunikkk) [reply gradle/wrapper/gradle-wrapper.properties:3]: Well, this is gradle used to build the project, but I'll drop a comment about other dependency updates

File: gradle/dependencies.gradle (hunk 2 above)
Comment (yunikkk) [reply gradle/dependencies.gradle:3]: Well, ok

File: CHANGELOG.md (hunk 5 above)
Comment (yunikkk) [reply CHANGELOG.md:3]: Scheduled for tomorrow

File: CHANGELOG.md (hunk 1 above)
Comment (yunikkk) [reply CHANGELOG.md:6]: Added link to the https://docs.mapbox.com/android/maps/guides/user-interaction/ at https://github.com/mapbox/mapbox-gestures-android/commit/49e74da993e2324257557379a002c381a5e9bfa4#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R17

**Review by kiryldz — APPROVED** [review]

**Comment by pengdev** [comment]
> The only detector that passes nullable MotionEvents that were marked NonNull was the SimpleStandardOnGestureListener.
> What kind of validation do you have in mind?

I meant in the https://github.com/mapbox/mapbox-gestures-android/pull/95, we have added the NonNull annotation to for example, `SimpleStandardOnScaleGestureListener`,`SimpleStandardOnShoveGestureListener`, but indeed it seems the `SimpleStandardOnGestureListener`.

**Comment by yunikkk** [comment]
> 

Yes, in the other Listeners like `SimpleStandardOnShoveGestureListener` NonNull annotations is only used for the Detector paremeter eg `@NonNull ShoveGestureDetector detector` and passed argument is always `this` so could not be null.

**Pull request merged**